pkg/objectio/reader.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objectio
+
+import (
+	"context"
+	"github.com/matrixorigin/matrixone/pkg/fileservice"
+)
+
+type ObjectReader struct {
+	object *Object
+	name   string
+}
+
+func NewObjectReader(name string, fs fileservice.FileService) (Reader, error) {
+	reader := &ObjectReader{
+		name:   name,
+		object: NewObject(name, fs),
+	}
+	return reader, nil
+}
+
+func (r *ObjectReader) ReadMeta(extent Extent) (*Block, error) {
+	var err error
+	meta := &fileservice.IOVector{
+		FilePath: r.name,
+		Entries:  make([]fileservice.IOEntry, 1),
+	}
+	meta.Entries[0] = fileservice.IOEntry{
+		Offset: int(extent.offset),
+		Size:   int(extent.Length()),
+	}
+	err = r.object.fs.Read(context.Background(), meta)
+	if err != nil {
+		return nil, err
+	}
+	block := &Block{}
+	err = block.UnMarshalMeta(meta.Entries[0].Data)
+	if err != nil {
+		return nil, err
+	}
+	return block, err
+}
+
+func (r *ObjectReader) Read(extent Extent, idxs []uint16) (*fileservice.IOVector, error) {
```

Comment:
I am not sure this is the right API.   Most likely someone has done read meta, and pass down the meta here to get the real data.